src/k8s/pkg/utils/hostname_test.go

```diff
@@ -0,0 +1,38 @@
+package utils_test
+
+import (
+	"testing"
+
+	"github.com/canonical/k8s/pkg/utils"
+	. "github.com/onsi/gomega"
+)
+
+func TestCleanHostname(t *testing.T) {
+	for _, tc := range []struct {
+		hostname       string
+		expectHostname string
+		expectValid    bool
+	}{
+		{hostname: "w1", expectHostname: "w1", expectValid: true},
+		{hostname: "w1.internal", expectHostname: "w1.internal", expectValid: true},
+		{hostname: "w1.test.domain", expectHostname: "w1.test.domain", expectValid: true},
+		{hostname: "w1-with-dash", expectHostname: "w1-with-dash", expectValid: true},
+		{hostname: "Capital", expectHostname: "capital", expectValid: true},
```

Comment:
Should we really try to be smart with that?
We do some magic behind the scenes that might confuse the user as the name is different for different commands.
```
ubuntu@alice:~/k8s$ sudo k8s join-cluster <token> --name BadName
Joining the cluster. This may take some time, please wait.
Joined the cluster as "BadName".
Please allow some time for Kubernetes node registration.
ubuntu@alice:~/k8s$ sudo k8s status
status: ready.
high-availability: no

control-plane nodes:
  badname: 10.84.220.237
  peter: 10.84.220.158
```

Reply:
Kubernetes already does this behind the scenes. We could revisit if we see that it's confusing